src/internal.rs

```diff
@@ -283,14 +288,30 @@ impl Local {
         self.guard_count.set(guard_count - 1);
 
         if guard_count == 1 {
-            self.epoch.store(Epoch::starting(), Release);
+            self.epoch.store(Epoch::starting(), Ordering::Release);
 
             if self.handle_count.get() == 0 {
                 self.finalize();
             }
         }
     }
 
+    /// Unpins and then pins the `Local`.
+    #[inline]
+    pub fn repin(&self) {
+        let guard_count = self.guard_count.get();
+
+        // Update the local epoch only if there's only one guard.
+        if guard_count == 1 {
+            let epoch = self.epoch.load(Ordering::Relaxed);
+            let global_epoch = self.global().epoch.load(Ordering::Relaxed);
+
+            if epoch != global_epoch {
+                self.update_epoch(epoch, global_epoch, Ordering::Release);
```

Comment:
A `SeqCst` fence isn't required here, a `Release` store should be enough to ensure correct ordering.

- We need to ensure any loads/stores from the previous epoch do not leak into the new one.
- However we don't care if any loads/store from the new epoch happen before the epoch counter is updated. At worse , other threads will see us in an older epoch and will delay GC slightly.

Reply:
@Amanieu you're right.  Thanks for pointing out that!  Also it seems I forgot what I said a month ago: https://github.com/crossbeam-rs/rfcs/issues/18#issuecomment-336768214  😭